integration-tests/src/test/scala/tech/beshu/ror/integration/suites/DataStreamApiSuite.scala

```diff
@@ -246,113 +279,248 @@ trait DataStreamApiSuite extends AnyFreeSpec
 
             val dsm = new DataStreamManager(user1Client)
-            val response = dsm.getAllDataStreams()
-            response.responseCode should be(401)
+            val response = dsm.getDataStream(dataStream1)
+            response.responseCode should be(200)
+            response.responseJson("data_streams").arr.map(v => v("name").str).toSet should be(Set(dataStream1))
+          }
+        }
+        "with indices rule should" - {
+          "allow to get data stream when" - {
+            "the data stream name does match the allowed data stream names" excludeES(allEs6x, allEs7xBelowEs79x) in {
+              val dataStream = DataStreamNameGenerator.next("test")
+              createDataStream(dataStream, IndexTemplateNameGenerator.next)
+
+              val dsm = new DataStreamManager(user3Client)
+              val response = dsm.getDataStream(dataStream)
+              response.responseCode should be(200)
+              response.responseJson("data_streams").arr.map(v => v("name").str).toSet should be(Set(dataStream))
+            }
+          }
+          "return empty data streams when" - {
+            "the data stream name does not match the allowed data stream names" excludeES(allEs6x, allEs7xBelowEs79x) in {
+              val dataStream = DataStreamNameGenerator.next("admin")
+              createDataStream(dataStream, IndexTemplateNameGenerator.next)
+
+              val dsm = new DataStreamManager(user3Client)
+              val response = dsm.getDataStream(dataStream)
+              response.responseCode should be(200)
+              response.responseJson("data_streams").arr.map(v => v("name").str).toSet should be(Set.empty)
+            }
           }
         }
       }
     }
-    "get data stream" - {
-      "without indices rule should" - {
-        "allow to get data stream" excludeES(allEs6x, allEs7xBelowEs79x) in {
-          val dataStream = DataStreamNameGenerator.next("admin")
-          createDataStream(dataStream, IndexTemplateNameGenerator.next)
-          val response = adminDataStreamManager.getDataStream(dataStream)
+    "get data stream stats" - {
+      "without data_streams rule should" - {
+        "allow to get data stream stats" excludeES(allEs6x, allEs7xBelowEs79x) in {
+          val dataStream1 = DataStreamNameGenerator.next("admin")
+          val dataStream2 = DataStreamNameGenerator.next("admin")
+          createDataStream(dataStream1, IndexTemplateNameGenerator.next)
+          createDocsInDataStream(dataStream1, 1)
+          createDataStream(dataStream2, IndexTemplateNameGenerator.next)
+          createDocsInDataStream(dataStream2, 2)
+          val response = adminDataStreamManager.getDataStreamStats("*")
           response.responseCode should be(200)
-          response.responseJson("data_streams").arr.map(v => v("name").str).toList should be(List(dataStream))
+          response.responseJson("data_stream_count").num.toInt should be(2)
```

Comment:
IMO, in general, we should avoid extracting JSON fields all over the tests. Let's create utility methods for this (in the manager).